### DISCUSSION TITLE
Cron style option for scheduling

### DISCUSSION THREAD

**Discussion by chris-reusch** [Miksus/rocketry#59] [Ideas]
Can you make the cron style scheduling an option for Rocketry? It would be much more precise and familiar.
ex: https://crontab.guru/

**Comment by Miksus** [comment]
I don't see it as a big problem. As far as I understand it, it's just the existing conditions but parsed from a bit different form.

I'm currently working on the async support (tests passed, need to write a bit documentation) and finalizing exposing more conditions to the condition API but after those I could take a look.

**Comment by Miksus** [comment]
UPDATE: This is now implemented. It should support the standard cron syntax (and a bit more, it's forgiving especially with week names etc.).

```python
from rocketry.conds import cron

@app.task(cron("5 4 * * *"))
def do_things():
    ...
``` 
Or specify the arguments:

```python
@app.task(cron(minute="*/2", hour="7-18", day_of_month="1,2,3", month="Feb-Aug/2"))
def do_things():
    ...
```

And this is just a regular condition which you can further combine with other conditions:

```python
from rocketry.conds import after_success

@app.task(cron("5 4 * * *") & after_success("another task"))
def do_things():
    ...
``` 

This fits nicely to the framework and it's implemented natively on the time period mechanics (which is actually complex but not really documented). Perhaps in the future we could have some sort of estimation of when tasks will run next time using the periods.
